src/addresses/address-details/AddressDetails.component.test.tsx

```diff
@@ -0,0 +1,139 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import { AddressDetails } from './AddressDetails.component';
+import { useJolokiaServiceReadAddressAttributes } from '../../openapi/jolokia/queries';
+import { AuthContext } from '../../jolokia/customHooks';
+
+jest.mock('../../openapi/jolokia/queries', () => ({
+  useJolokiaServiceReadAddressAttributes: jest.fn(),
+}));
+
+describe('AddressDetails', () => {
+  const mockData = [
+    { request: { attribute: 'Address' }, value: 'DLQ' },
+    { request: { attribute: 'AddressSize' }, value: 0 },
+    { request: { attribute: 'AutoCreated' }, value: false },
+    { request: { attribute: 'MessageCount' }, value: 0 },
+    { request: { attribute: 'RoutingTypes' }, value: 'Anycast' },
+    { request: { attribute: 'BindingNames' }, value: 'DLQ' },
+    { request: { attribute: 'CurrentDuplicateIdCacheSize' }, value: 0 },
+    { request: { attribute: 'Internal' }, value: false },
+    { request: { attribute: 'AddressLimitPercent' }, value: 0 },
+    { request: { attribute: 'NumberOfBytesPerPage' }, value: 1245 },
+    { request: { attribute: 'NumberOfMessages' }, value: 0 },
+    { request: { attribute: 'QueueNames' }, value: 'DLQ' },
+  ];
+
+  beforeEach(() => {
+    (useJolokiaServiceReadAddressAttributes as jest.Mock).mockReturnValue({
+      data: mockData,
+      isSuccess: true,
+      error: null,
+    });
+  });
+
+  it('should renders data correctly when fetch is successful', () => {
+    render(
+      <AuthContext.Provider value="mock-token">
+        <AddressDetails name="DLQ" />
+      </AuthContext.Provider>,
+    );
+
+    expect(screen.getByText('Address')).toBeInTheDocument();
+    expect(screen.getByText('AddressSize')).toBeInTheDocument();
+    expect(screen.getByText('AutoCreated')).toBeInTheDocument();
+    expect(screen.getByText('MessageCount')).toBeInTheDocument();
+    expect(screen.getByText('RoutingTypes')).toBeInTheDocument();
+  });
+
+  it('should renders loading spinner initially when data not available', () => {
+    (useJolokiaServiceReadAddressAttributes as jest.Mock).mockReturnValueOnce({
+      data: null,
+      isSuccess: false,
+      error: null,
+    });
+
+    render(
+      <AuthContext.Provider value="mock-token">
+        <AddressDetails name="DLQ" />
+      </AuthContext.Provider>,
+    );
+
+    expect(screen.getByText('attributes')).toBeInTheDocument();
```

Comment:
Here should it be checking the attribute values are empty or absent in case of read failure?